packages/beacon-node/src/chain/validation/attestation.ts

```diff
@@ -161,6 +164,161 @@ export async function validateGossipAttestation(
+    const aggregationBits = attestation.aggregationBits;
+    const bitIndex = aggregationBits.getSingleTrueBit();
+    if (bitIndex === null) {
+      throw new AttestationError(GossipAction.REJECT, {
+        code: AttestationErrorCode.NOT_EXACTLY_ONE_AGGREGATION_BIT_SET,
+      });
+    }
+
+    // [REJECT] The committee index is within the expected range
+    // -- i.e. data.index < get_committee_count_per_slot(state, data.target.epoch)
+    const attIndex = attData.index;
+    const committeeIndices = getCommitteeIndices(attHeadState, attSlot, attIndex);
+
+    const validatorIndex = committeeIndices[bitIndex];
+    validatorIndices.push(validatorIndex);
+
+    // [REJECT] The number of aggregation bits matches the committee size
+    // -- i.e. len(attestation.aggregation_bits) == len(get_beacon_committee(state, data.slot, data.index)).
+    // > TODO: Is this necessary? Lighthouse does not do this check
+    if (aggregationBits.bitLen !== committeeIndices.length) {
+      throw new AttestationError(GossipAction.REJECT, {
+        code: AttestationErrorCode.WRONG_NUMBER_OF_AGGREGATION_BITS,
+      });
+    }
+
+    // api attestations don't have subnet so never throw AttestationErrorCode.INVALID_SUBNET_ID
+    const expectedSubnet = attHeadState.epochCtx.computeSubnetForSlot(attSlot, attIndex);
+
+    // [IGNORE] There has been no other valid attestation seen on an attestation subnet that has an
+    // identical attestation.data.target.epoch and participating validator index.
+    if (chain.seenAttesters.isKnown(targetEpoch, validatorIndex)) {
+      throw new AttestationError(GossipAction.IGNORE, {
+        code: AttestationErrorCode.ATTESTATION_ALREADY_KNOWN,
+        targetEpoch,
+        validatorIndex,
+      });
+    }
+
+    // [REJECT] The signature of attestation is valid.
+    const indexedAttestation: phase0.IndexedAttestation = {
+      attestingIndices: [validatorIndex],
+      data: attData,
+      signature: attestation.signature,
+    };
+
+    signatureSets.push(getIndexedAttestationSignatureSet(attHeadState, indexedAttestation));
+
+    // Now that the attestation has been fully verified, store that we have received a valid attestation from this validator.
+    //
+    // It's important to double check that the attestation still hasn't been observed, since
+    // there can be a race-condition if we receive two attestations at the same time and
+    // process them in different threads.
+    if (chain.seenAttesters.isKnown(targetEpoch, validatorIndex)) {
+      throw new AttestationError(GossipAction.IGNORE, {
+        code: AttestationErrorCode.ATTESTATION_ALREADY_KNOWN,
+        targetEpoch,
+        validatorIndex,
+      });
+    }
+
+    validationResults.push({indexedAttestation, subnet: expectedSubnet});
+  }
+
+  if (!(await chain.bls.verifySignatureSets(signatureSets, {batchable: true}))) {
+    throw new AttestationError(GossipAction.REJECT, {code: AttestationErrorCode.INVALID_SIGNATURE});
+  }
```

Comment:
The behavior as implemented here will cause all attestations to be rejected if only of the them is wrong. This is not correct, and while you can argue that an APIi is _likely_ to submit attestations that are all valid this is not an assumption I'm comfortable taking. If you want to go the route of batchable verify functions I did an implementation in this branch which supports individual status reporting https://github.com/ChainSafe/lodestar/blob/dapplion/attestation-batch/packages/beacon-node/src/chain/validation/attestation.ts

Reply:
if it fails here, at the `submitPoolAttesatations()` api, it'll fallback to verify each attestation separately and will publish only valid attestation as the old flow https://github.com/ChainSafe/lodestar/pull/5311/files#diff-2381400ee782266412ea40f4ba10a08e49c7e60a3b63dbacaa8774d9ae583fcfR111

I should add some comments to this line